Yosemite/Yosemite/Stores/OrderStore.swift

```diff
@@ -370,6 +372,20 @@ private extension OrderStore {
             }
         }
     }
+
+    /// Deletes a given order.
+    ///
+    func deleteOrder(siteID: Int64, orderID: Int64, deletePermanently: Bool, onCompletion: @escaping (Result<Order, Error>) -> Void) {
+        remote.deleteOrder(for: siteID, orderID: orderID, force: deletePermanently) { [weak self] result in
+            switch result {
+            case .success:
+                self?.deleteStoredOrder(siteID: siteID, orderID: orderID)
```

Comment:
[Q] I wonder if we should delete it locally first, and re-add it if the network operation failed. 🤔

Reply:
That's a good thought — it would be more optimistic. I made that change in [512f6ec](https://github.com/woocommerce/woocommerce-ios/pull/6395/commits/512f6ec4094c6a0ef71b54194876bb9d7c3c77e4) (passing in the full order instead of the order ID to make it easier to re-add); what do you think?

Reply:
Looks great!